internal/app/api/handler/login/login.go

```diff
@@ -0,0 +1,91 @@
+package login
+
+import (
+	"fmt"
+	"github.com/TheFlies/ofriends/internal/app/ldap"
+	"github.com/TheFlies/ofriends/internal/app/types"
+	"github.com/TheFlies/ofriends/internal/pkg/glog"
+	"github.com/TheFlies/ofriends/internal/pkg/jwt"
+	"github.com/TheFlies/ofriends/internal/pkg/respond"
+	"github.com/globalsign/mgo/bson"
+	"github.com/sirupsen/logrus"
+	"net/http"
+	"time"
+)
+
+type (
+	service interface {
+		GetUserByeUserName(username string) (*types.Usercache, error)
+		AddUser(user *types.Usercache) error
+		CheckUseExist(username string) bool
+	}
+	LoginHandler struct {
+		srv    service
+		logger glog.Logger
+	}
+)
+
+func NewLoginHandeler(s service, l glog.Logger) *LoginHandler {
+	return &LoginHandler{
+		srv:    s,
+		logger: l,
+	}
+}
+func (h *LoginHandler) Authentication(w http.ResponseWriter, r *http.Request) {
+	usename := r.FormValue("username")
+	password := r.FormValue("password")
+	err := ldap.LDAPAuthentication(usename, password)
```

Comment:
The Login handler should use a login/user service and the service can have multiple authenticators, one of them is LDAP. Hence these kind of logic should be handled by the service layer, not the handler.